### PR TITLE
parser: fix self-referenced error (fix #5013)

### LINKED ISSUE #5013: Crash self-referenced
**V version:** V 0.1.27 4189190
**OS:** macOS 10.15

**What did you do?**
```
$ cat inf.v
fn main()
{
	s := '$s'
	eprintln(s)
}
$ v inf.v
Segmentation fault: 11
$
```

**What did you expect to see?**

an error because 's' is not yet defined at this time
 
**What did you see instead?**
Segmentation fault: 11

### PR DESCRIPTION
**Pull request by yuyi98**
This PR fix self-referenced error (fix #5013).

- Fix self-referenced error.
- Add test `assign_expr_unresolved_variables_err_d.vv/out`.

```v
D:\test\v\tt1>v run .
.\tt1.v:3:9: error: unresolved variables `s`
    1 | fn main()
    2 | {
    3 |     s := '$s'
      |            ^
    4 |     println(s)
    5 | }
```

### CODE REVIEW

**Comment by JalonSolov** [comment]
That error message does not look correct.  `s` is resolved, as it is the variable you want to assign to.

The problem is that the code is attempting to assign an undefined value to `s`.

**Comment by yuyi98** [comment]
I think it's unresolved before assign.
I don't know if it's appropriate to use `undefined` or  `unresolved`.

**Comment by Delta456** [comment]
`undefined` is better here.

**Comment by yuyi98** [comment]
OK, I'll change it.

File: vlib/v/checker/tests/assign_expr_unresolved_variables_err_a.out [hunk 1]
```diff
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/assign_expr_unresolved_variables_err_a.v:2:7: error: unresolved variables `a`
+vlib/v/checker/tests/assign_expr_unresolved_variables_err_a.v:2:7: error: undefined variables `a`
```
Comment (medvednikov) [inline vlib/v/checker/tests/assign_expr_unresolved_variables_err_a.out:1]:
```
undefined: `a`
```

File: vlib/v/checker/tests/assign_expr_unresolved_variables_err_b.out [hunk 2]
```diff
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/assign_expr_unresolved_variables_err_b.v:2:10: error: unresolved variables `a`
+vlib/v/checker/tests/assign_expr_unresolved_variables_err_b.v:2:10: error: undefined variables `a`
```
Comment (Delta456) [inline vlib/v/checker/tests/assign_expr_unresolved_variables_err_b.out:1]: Change filename to `undefined`

File: vlib/v/checker/tests/assign_expr_unresolved_variables_err_d.vv [hunk 3]
```diff
@@ -0,0 +1,4 @@
+fn main() {
+	s := '$s'
+	println(s)
+}
```
Comment (Delta456) [inline vlib/v/checker/tests/assign_expr_unresolved_variables_err_d.vv:4]: Newline here

**Pull request merged**